### PR TITLE
Amir/sam projects2 pr3

### PR DESCRIPTION
**Pull request by amir-ghasemi**
# Projects Feature: Default Agent, Advanced Search & Project Management

## Overview

This PR (PR3) builds on PR2 and introduces default agent configuration, advanced search capabilities, comprehensive project management tools, and critical bug fixes to the projects feature.

### 1. 🎯 Default Agent Configuration (f4e31ba)

Added ability to configure a default agent per project that automatically applies to new chat sessions:

**Database Changes:**
- New `default_agent_id` column in projects table (nullable, foreign key to agents)
- Alembic migration: `20251024_add_default_agent_to_projects.py`
- Updated Project entity, model, and repository to handle default agent

**Backend Changes:**
- Enhanced ProjectService with default agent validation and management
- Updated project DTOs (CreateProjectRequest, UpdateProjectRequest, ProjectResponse)
- Modified project creation/update endpoints to accept and return default_agent_id
- Added default agent to project API responses

**Frontend Changes:**
- New `DefaultAgentSection` component (116 lines) with agent selection dropdown
- Integrated into ProjectMetadataSidebar for easy configuration
- Updated ChatProvider to automatically use project's default agent when creating sessions
- Added default_agent_id to TypeScript project types

**Testing:**
- Updated integration test fixtures (conftest.py)
- Added test coverage in declarative runner scenarios
- All tests passing

**Impact:** Streamlines workflow by eliminating need to manually select agent for each new chat in a project

---

### 2. 🔍 Advanced Chat Search (39cd2e9)

Implemented full-text search across session titles and chat content:

**Database Changes:**
- Added full-text search indexes on sessions table (title column)
- Added full-text search indexes on chat_tasks table (content column)
- Alembic migration: `20251023_add_fulltext_search_indexes.py`
- Optimized indexes for search performance

**Backend Changes:**
- Enhanced SessionRepository with search capabilities
- Added search query support in session service
- Updated session endpoints to accept search parameters

**Frontend Changes:**
- Search input in session list
- Real-time search filtering
- Highlighted search results

**Impact:** Dramatically improves discoverability of conversations within projects, especially for projects with many sessions

---

### 3. 🛠️ Project Management Enhancements (97d70d2)

Added comprehensive project management tools including move chats, cascade delete, and enhanced search:

**Move Chat Between Projects:**
- New `MoveSessionDialog` component for UI
- Backend endpoint to move sessions between projects
- Validation to ensure user has access to both projects
- Updates session's project_id and maintains history

**Cascade Delete:**
- Enhanced project deletion to cascade to sessions and tasks
- Prevents orphaned data in database
- Proper cleanup of all related records

**Enhanced Search:**
- Project-level search across all sessions
- Combined with full-text search from commit #2
- Filter and sort capabilities

**Files Modified:**
- Backend: project_repository.py, project_service.py, projects.py router
- Frontend: MoveSessionDialog.tsx, SessionList.tsx, ProjectMetadataSidebar.tsx

**Impact:** Provides users with professional-grade project management capabilities

---

### 4. 🏷️ Filter Badges Row (954a999)

Improved project badge display and filtering:

**Changes:**
- Enhanced badge filtering logic in session history
- Better visual organization of project badges
- Improved badge styling and layout
- Responsive design for badge rows

**Impact:** Cleaner UI with better visual hierarchy for project metadata

---

### 5. 🐛 Session Title Edit Bug Fix (4b5bbee)

Fixed critical bug where editing the first session title wasn't reflected in UI:

**Root Cause:**
- State synchronization issue between session list and active session
- First session had special handling that bypassed normal update flow

**Fix:**
- Corrected state update logic in SessionList component
- Ensured proper event propagation for title changes
- Added proper re-rendering triggers

**Impact:** Resolves user-reported bug affecting session management

---

### 6. 🔧 TypeScript Error Resolution (d7845f6)

Fixed TypeScript compilation errors in ChatPage and ChatProvider:

**Changes:**
- Removed unused variables in ChatPage
- Added explicit type annotations in ChatProvider
- Fixed type mismatches in project context handling
- Ensured strict type safety

---

### 7. ✅ Initial Test Suite (2ae6e66)

Added comprehensive test coverage for project API endpoints:

**Test Coverage:**
- Project CRUD operations
- Multi-user access scenarios
- Feature flag behavior
- Error handling and edge cases

**Test Files:**
- `test_data/projects/test_create_project_success.yaml`
- Additional declarative test scenarios

**Impact:** Ensures reliability and prevents regressions

### New Features
1. **Default Agent Configuration** - Projects can specify default agent for new sessions
2. **Advanced Search** - Full-text search across session titles and chat content
3. **Move Chats** - Transfer sessions between projects
4. **Cascade Delete** - Proper cleanup when deleting projects
5. **Enhanced Filtering** - Better badge organization and filtering

### Bug Fixes
1. Session title editing for first session
2. TypeScript compilation errors
3. Project ordering (newest first)

### Database Changes
- New `default_agent_id` column in projects table
- Full-text search indexes on sessions and chat_tasks
- Soft delete support (from PR2)
- Performance optimization indexes

## Testing

- All TypeScript compilation errors resolved ✅
- Integration tests updated and passing ✅
- Database migrations tested ✅
- Manual UI testing completed ✅
- No breaking changes ✅

## Notes

- This PR builds on top of PR2 (`amir/sam-projects2-pr2` branch)
- All changes are backward compatible
- Feature flags from PR2 remain functional
- Multi-user infrastructure from PR2 is utilized

### CODE REVIEW

File: client/webui/frontend/src/lib/components/projects/DefaultAgentSection.tsx [hunk 1]
```diff
@@ -0,0 +1,116 @@
...
+                    </div>
+                ) : (
+                    <div className="text-sm text-muted-foreground rounded-md bg-muted p-3 min-h-[60px] flex items-center">
+                        {project.defaultAgentId ? (
+                            <div className="flex items-center gap-2">
+                                <Bot className="h-4 w-4" />
+                                <span>{displayName}</span>
+                            </div>
+                        ) : (
+                            <span className="italic">No default agent set. New chats will prompt for agent selection.</span>
```
Comment (RohanSreelesh) [inline client/webui/frontend/src/lib/components/projects/DefaultAgentSection.tsx:109]: I don't think this is the case here. we can just say no default agent set for now
